### ISSUE TITLE
Should pflogger have an option for enabling/disabling MPI

### ISSUE THREAD

**Issue by climbfuji** [Goddard-Fortran-Ecosystem/pFlogger#106]
This question is related to issue https://github.com/spack/spack/issues/42137.

I think it is problematic if MPI is enabled silently if it's found somewhere in the search path. That MPI might not be compatible with the compiler/environment that the user is trying to build pflogger for. Wouldn't it be better to make this an option that is off by default and needs to be enabled explicitly, and then fail if enabled but no MPI can be found?

**Comment by mathomp4** [comment]
Yeah, I suppose having something like an `ENABLE_MPI` switch that defaults to `ON` would be safe. Then past behavior is preserved.

**Comment by climbfuji** [comment]
That would work for us, thanks!

**Comment by tclune** [comment]
@mathomp4 Could you make the attempt, and I'll review?

**Comment by mathomp4** [comment]
> @mathomp4 Could you make the attempt, and I'll review?

Okay. I'll give it a shot.

**Comment by mathomp4** [comment]
I've a PR in #107 which I think does this. 

(Note that the CI might not quite work (old GCC?) so I've updated #94 to be like the CI in pFUnit develop. Hopefully this will work, but it might not as that has ifx...)
